### ISSUE TITLE
[FR] Time tracking plugin

### ISSUE THREAD

**Issue by hgkamath** [AppFlowy-IO/AppFlowy#2629]
### Description

An interface wherein one can 
- create projects (perhaps import this from a task planner/scheduler)
- create tasks in each projects (perhaps import this from a task planner/scheduler)
- provide a view to select/start/stop an ongoing project/task currently being worked on. 
  - perhaps records activity to a calendar/planner, optionally record a comment on the task. 
- provide a view to select a date/time and visualize the allocation of time per project in a day-range/week-range/month-range/year-range , either as a pie chart-or stacked-bar chart.

### Impact

Users who need a time tracking system.
One can't improve what one does not measure.   
A time tracker lets one be honest to oneself about how long some tasks take. 

### Additional Context

Maybe similar to 
ex: https://github.com/kromitgmbh/titra/

**Comment by Jabe64** [comment]
A great inspiration for Time Tracking and productivity options is Super Productivity : 
- [https://github.com/johannesjo/super-productivity](https://github.com/johannesjo/super-productivity)
- [https://super-productivity.com](https://super-productivity.com/)

There is a [video demo here](https://www.youtube.com/watch?v=KUqRFSMIXck).

Also you can get inspiration from [Monday.com](https://www.monday.com/) but this one isn't entirely free.

I think if AppFlowy incoporates such features a lot of people will adopt it instead of other collaborative and productivity tools.

**Comment by naikymen** [comment]
+1 to a stopwatch button as found in SuperProductivity. It is a simple feature, and provides very valuable information for productivity.

For implementation I'd recommend porting these features from SP:

- Adding a small "⏯" button to each task in a kanban view that has a time tracking property.
- Add a pop-up that asks whether you have been working (or not) on the active task when the app is idle for some time.

It would also be possible to sync SP to AppFlowy through an API, as it already implements sync and time tracking with other platforms (e.g. GitLab issues).

**Comment by Jabe64** [comment]
Yes I agree and another inspiration could be Clickup Time Tracking:
[https://clickup.com/features/project-time-tracking](https://clickup.com/features/project-time-tracking)

**Comment by hadifarnoud** [comment]
it has been a while, is this feature planned?
